lib/src/files.rs

```diff
@@ -269,142 +269,22 @@ mod tests {
 
     #[test]
     fn test_merge() {
-        // Unchanged and empty on all sides
```

Comment:
> Most of the complexity is now in `trivial_merge()`, so I think we can
> remove most of the tests.
 
FWIW I think it's worth keeping the tests because it makes it easier to change this file to not use `trivial_merge()`. It's probably also worth checking that `trivial_merge()` is being used correctly.